File: CopilotDemoApp.Server/Features/Order/Order.cs
using CopilotDemoApp.Server.Database;

namespace CopilotDemoApp.Server.Features.Order;

public sealed record Order(
	Guid Id,
	string UserId,
	string UserEmail,
	string ShippingAddress,
	string ShippingCity,
	string ShippingProvince,
	string ShippingPostalCode,
	DateTime OrderDate,
	OrderStatus Status,
	decimal TotalAmount,
	List<OrderLineItem> LineItems
);

File: frontend/src/types/order.ts
export const OrderStatus = {
  Pending: 0,
  Processing: 1,
  Shipped: 2,
  Delivered: 3,
} as const;

export type OrderStatus = (typeof OrderStatus)[keyof typeof OrderStatus];

export interface OrderLineItem {
  productId: string;
  productName: string;
  productPrice: number;
  quantity: number;
}

export interface Order {
  id: string;
  userId: string;
  userEmail: string;
  shippingAddress: string;
  shippingCity: string;
  shippingState: string;
  shippingPostalCode: string;
  orderDate: string;
  status: OrderStatus;
  totalAmount: number;
  lineItems: OrderLineItem[];
}

export interface CreateOrderLineItemDto {
  productId: string;
  productName: string;
  productPrice: number;
  quantity: number;
}

export interface CreateOrderRequest {
  shippingAddress: string;
  shippingCity: string;
  shippingState: string;
  shippingPostalCode: string;
  lineItems: CreateOrderLineItemDto[];
  cardNumber: string;
  cardholderName: string;
  expiryDate: string;
  cvv: string;
}
